pstress/generate_rocksdb_options.sh

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash                                                                         #
+# Created by Mohit Joshi, Percona LLC                                               #
+# Created on: 06-SEP-2021                                                           #
+#                                                                                   #
+# Purpose: This script auto-generates the rocksdb_options.txt file                  #
+# Usage: ./generate_rocksdb_options.sh <path to mysql install/base dir>             #
+# References: https://www.percona.com/doc/percona-server/8.0/myrocks/variables.html #
+#####################################################################################
+
+# Script expected argument
+BASEDIR=$1
+
+# User variables
+OUTPUT_FILE=/tmp/rocksdb_options_80.txt
+
+# Internal variables, do not change
+TEMP_FILE=/tmp/rocksdb_options_80.tmp
+PORT=22000
+SOCKET=/tmp/mysql_22000.sock
+DATADIR=$BASEDIR/data
+MYSQLD_START_TIMEOUT=30
+
+# Check if mysqld binaries exists
+if [ ! -r $BASEDIR/bin/mysqld ]; then
+    echo "Error: no ./bin/mysqld or ./mysqld found!"
+    exit 1
+fi
+
+# Display function
+echoit(){
+  echo "[$(date +'%T')] $1"
+  echo "[$(date +'%T')] $1" >> /tmp/generate_rocksdb_options.log
+}
+
+echoit "Starting MySQL server"
+$BASEDIR/bin/mysqld --no-defaults --datadir=$DATADIR --initialize-insecure > /tmp/mysql_install_db.txt 2>&1
+$BASEDIR/bin/mysqld --no-defaults --datadir=$DATADIR --port=$PORT --socket=$SOCKET 2>&1 > /tmp/master.err 2>&1 &
+MPID=$!
+
+echoit "Waiting for the server to fully start..."
+for X in $(seq 0 ${MYSQLD_START_TIMEOUT}); do
+  sleep 1
+  if [ "$MPID" == "" ]; then echoit "Assert! ${MPID} empty. Terminating!"; exit 1; fi
+  if $BASEDIR/bin/mysqladmin -uroot -S${SOCKET} ping > /dev/null 2>&1; then
+    echoit "MySQL server started successfully"
+    $BASEDIR/bin/ps-admin --enable-rocksdb -uroot -S/tmp/mysql_22000.sock 2>&1 > /dev/null
+    echoit "RocksDB loaded successfully"
+    break
+  fi
+done
+
+# Extract all options, their default values into temp file
+$BASEDIR/bin/mysql --no-defaults -S$SOCKET -uroot -e "SHOW GLOBAL VARIABLES LIKE '%rocksdb%'" > $TEMP_FILE
+
+# Remove the first line from the temp file
+# Variable_name, Variable_value (we do not need the column name displayed in temp file)
+sed -i 1d $TEMP_FILE
+
+# List of rocksdb variables which must not be changed.
+EXCLUDED_LIST=( --loose-rocksdb_compact_cf --loose-rocksdb_default_cf_options --loose-rocksdb_update_cf_options --loose-rocksdb_delete_cf --loose-rocksdb_override_cf_options --loose-rocksdb_create_checkpoint --loose-rocksdb_create_temporary_checkpoint --loose-rocksdb_fault_injection_options --loose-rocksdb_read_free_rpl_tables --loose-rocksdb_persistent_cache_path --loose-rocksdb_strict_collation_exceptions --loose-rocksdb_tmpdir --loose-rocksdb_trace_block_cache_access --loose-rocksdb_trace_queries --loose-rocksdb_wal_dir --loose-rocksdb_wsenv_path --loose-rocksdb_datadir )
```

Comment:
`rocksdb_default_cf_options` is one of most important options.
Using this option you can bypass MyRocks and provide options directly to RocksDB.
Please check https://github.com/facebook/mysql-5.6/wiki/my.cnf-tuning for some examples

Reply:
Fixed